scripts/test_solidity_lsp.py

```diff
@@ -0,0 +1,379 @@
+            [self.exe_path, *self.exe_args],
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback) -> None:
+        self.process.kill()
+        self.process.wait(timeout=2.0)
+
+    def trace(self, topic: str, message: str) -> None:
+        if self.trace_io:
+            print(f"{SGR_TRACE}{topic}:{SGR_RESET} {message}")
+
+    def receive_message(self) -> Any:
+        # `, timeout: float = 2.0`
+        # Note, we should make use of timeout to avoid infinite blocking if nothing is received.
+        LEN_HEADER = "Content-Length: "
+        TYPE_HEADER = "Content-Type: "
+        if self.process.stdout == None:
+            return None
+        message_size = None
+        while True:
+            #read header
+            line = self.process.stdout.readline()
+            if not line:
+                # server quit
+                return None
+            line = line.decode("utf-8")
+            if not line.endswith("\r\n"):
+                raise RuntimeError("Bad header: missing newline")
+            # remove the "\r\n"
+            line = line[:-2]
+            if line == "":
+                break # done with the headers
+            if line.startswith(LEN_HEADER):
+                line = line[len(LEN_HEADER):]
+                if not line.isdigit():
+                    raise RuntimeError("Bad header: size is not int")
+                message_size = int(line)
+            elif line.startswith(TYPE_HEADER):
+                # nothing todo with type for now.
+                pass
+            else:
+                raise RuntimeError("Bad header: unknown header")
+        if not message_size:
+            raise RuntimeError("Bad header: missing size")
+        rpc_message = self.process.stdout.read(message_size).decode("utf-8")
+        self.trace('receive_message', rpc_message)
+        return json.loads(rpc_message)
+
+    def send_message(self, method_name: str, params: Any) -> None:
+        if self.process.stdin == None:
+            return
+        message = { 'jsonrpc': '2.0', 'method': method_name, 'params': params }
+        json_string = json.dumps(obj=message, cls=MyEncoder)
+        rpc_message = f"Content-Length: {len(json_string)}\r\n\r\n{json_string}"
+        self.trace(f'send_message ({method_name})', json_string)
+        self.process.stdin.write(rpc_message.encode())
+        self.process.stdin.flush()
+
+    def call_method(self, method_name: str, params: Any) -> Any:
+        self.send_message(method_name, params)
+        return self.receive_message()
+
+    def send_notification(self, name: str, params: Any = None) -> None:
+        self.send_message(name, params)
+
+# }}}
+
+SGR_RESET = '\033[m'
+SGR_TRACE = '\033[1;36m'
+SGR_TEST_BEGIN = '\033[1;33m'
+SGR_ASSERT_BEGIN = '\033[1;34m'
+SGR_STATUS_OKAY = '\033[1;32m'
+SGR_STATUS_FAIL = '\033[1;31m'
```

Comment:
Maybe we could use a library for this? I think this won't work in a Windows terminal and overall we could use colors in more Python scripts but hard-coding them everywhere is not a good idea.

How about e.g. [colorama](https://github.com/tartley/colorama)?

Reply:
using colorama now.